### PR TITLE
RKE2 tag release scripts

### PR DESCRIPTION
**Pull request by jossemargt-3pillar**
This PR tackles multiple scenarios at once, which are:

- Enable RKE2 tag release script to determine when to tag against `master` or a `release-1.2x` branch
- Refactor tag related scripts to use the same "dry run" mechanism
- Rename tag related scripts using the same name pattern `tag_<component>_release`

Each scenario comes on its own commit, so in case we don't want said behavior we can drop it.

### CODE REVIEW

File: bin/tag_image_build_k8s_release [hunk 1]
```diff
@@ -55,15 +57,11 @@ DATE=$(TZ=utc date '+%Y%m%d')
...
-    if [ ${DRY_RUN} ]; then
-        echo "gh release create ${TAG} --repo rancher/image-build-kubernetes --title ${TAG} --target master --notes ''"
-    else
-        gh release create "${TAG}"                  \
-            --repo "rancher/image-build-kubernetes" \
-            --title "${TAG}"                        \
-            --target "master"                       \
-            --notes ""
-    fi
+    $DR_PREFIX gh release create "${TAG}"       \
```
Comment (briandowns) [inline bin/tag_image_build_k8s_release:60]: Add curly braces to var.

File: bin/tag_rke2_release [hunk 2]
```diff
@@ -4,28 +4,33 @@ set -e
 
 . libstd-ecm.sh
 
+DR_PREFIX=''
+MAIN_MINOR='23'
```
Comment (briandowns) [inline bin/tag_rke2_release:8]: I'm not sure we should be tightyly coupling to a version here.

Comment (jossemargt-3pillar) [reply bin/tag_rke2_release:8]: Even though you  can override that value with `-m` flag? 

An alternative that comes to my mind is to query rke2 master branch and look for that value in the `version.sh` wdyt?

Comment (briandowns) [reply bin/tag_rke2_release:8]: I guess I'd rather have the -m be required and be explicit.

**Review by nikolaishields — APPROVED** [review]

**Pull request merged**